hack/build_channel_catalog.sh

```diff
@@ -17,32 +26,27 @@ for var in BUNDLE_IMAGE \
            GOOS \
            GOARCH \
            OPM_VERSION \
-           BUNDLE_VERSIONS_REPO \
-           BRANCH_CHANNEL \
-           APP_INTERFACE_USERNAME \
-           APP_INTERFACE_PASSWORD \
-           APP_INTERFACE_BASE_URL
+           BUNDLE_VERSIONS_REPO
 do
-    if [ ! "${!var}" ]; then
+    if [ ! "${!var:-}" ]; then
       log "$var is not set"
       count=$((count + 1))
     fi
 done
 
 [[ $count -gt 0 ]] && exit 1
 
-# We shouln't need set -u as we have checked for the vars before
-# but let's play safe
-set -euo pipefail
-
-log "Building $OPERATOR_NAME version $OPERATOR_VERSION in branch $BRANCH_CHANNEL"
+# Check we are running an opm supported container engine
+image_builder=$(basename "$CONTAINER_ENGINE")
+if [[ "$image_builder" != "docker" && "$image_builder" != "podman" ]]; then
+    # opm error messages are obscure. Let's make this clear
+    log "image_builder $image_builder is not one of docker or podman"
+    exit 1
+fi
 
-# General vars to modify behaviour of the script
-DRY_RUN=${DRY_RUN:-false}
-DELETE_TEMP_DIR=${DELETE_TEMP_DIR:-true}
-REMOVE_UNDEPLOYED=${REMOVE_UNDEPLOYED:-false}
+log "Building $OPERATOR_NAME version $OPERATOR_VERSION"
 
-temp_dir=$(mktemp -d --suffix "-$(basename $0)")
+temp_dir=$(mktemp -d --suffix "-$(basename "$0")")
```

Comment:
Curious why this delta. The result will almost always be the same. In particular, you'll still break if there are spaces in `$0` (but why would there be?). Did you mean
```suggestion
temp_dir=$(mktemp -d --suffix "-$(basename '"$0"')")
```
?

Reply:
I was making `shellcheck` happy by quoting every single variable that has to be quoted

Reply:
except what you've done here is *un*quote `$0`.

Reply:
bash doesn't agree with you :)

```
[rporresm@rporresm-mac test]$ cat example\ with\ spaces.sh
temp_dir=$(mktemp -d --suffix "-$(basename "$0")")
echo "$temp_dir"
[rporresm@rporresm-mac test]$ bash example\ with\ spaces.sh
/var/folders/dx/y5_klhc1187gnzzmyb4pswb40000gn/T/tmp.buSMNjRKyQ-example with spaces.sh
```
I then remove the quote
```
[rporresm@rporresm-mac test]$ cat example\ with\ spaces.sh
temp_dir=$(mktemp -d --suffix "-$(basename $0)")
echo "$temp_dir"
[rporresm@rporresm-mac test]$ bash example\ with\ spaces.sh
basename: extra operand ‘spaces.sh’
Try 'basename --help' for more information.
/var/folders/dx/y5_klhc1187gnzzmyb4pswb40000gn/T/tmp.BugEWJylKo-
```

Reply:
Whoah.

I'm having a hard time following how bash is parsing that. But your example is convincing. I'll go away now.